### ISSUE TITLE
Unused block keyword arguments

### ISSUE THREAD

**Issue by bquorning** [bbatsov/rubocop#4698]
## Expected behavior

I want to use RSpec to implement a custom response block for a method that takes keyword arguments, without RuboCop warnings.

## Actual behavior

RuboCop tells me to omit the keyword argument. But since it‘s a named kwargs key, I can not just remove it, nor rename it.

## Steps to reproduce the problem

A test.rb file:

```rb
# frozen_string_literal: true

RSpec.describe 'bug' do
  it 'works' do
    foo = double
    allow(foo).to receive(:bar) { |baz:| 42 }
    expect(foo.bar(baz: 'hello world')).to eq(42)
  end
end
```

```
» rubocop test.rb
Inspecting 1 file
W

Offenses:

test.rb:6:36: W: Unused block argument - baz. You can omit the argument if you don't care about it.
    allow(foo).to receive(:bar) { |baz:| 42 }
                                   ^^^

1 file inspected, 1 offense detected
```

I know that this is a very contrived example, since 1) `foo` doesn’t have any original implementation of `bar`, and 2) `baz` is not used inside the block, so I could just use `and_return` instead. I went for “minimal reproduction case” over realism.

## RuboCop version

```
» rubocop -V
0.49.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin15)
```

**Comment by Drenmi** [comment]
I believe there has been a discussion on this before, but I might be wrong. I personally do not have enough hands-on experience with keyword argument blocks to have an opinion on this, so will defer. 😬

**Comment by bquorning** [comment]
> I believe there has been a discussion on this before

#3130 is related, but this is a different issue. But I see now that `UnusedBlockArgument` has an `AllowUnusedKeywordArguments` options that I perhaps should enable for my spec folder?

**Comment by Drenmi** [comment]
@bquorning: Sounds like it might. Give it a go, and see if it fits your case. If not we leave the issue open. 🙂

**Comment by bquorning** [comment]
That fixes it. The downside is that I now must have a separate rubocop config file for my specs, just to override that one option.
